src/shared/ExtensionMessage.ts

```diff
@@ -78,7 +78,7 @@ export interface ExtensionMessage {
 		| "promptsButtonClicked"
 		| "didBecomeVisible"
 	invoke?: "newChat" | "sendMessage" | "primaryButtonClick" | "secondaryButtonClick" | "setChatBoxMessage"
-	state?: ExtensionState
+	state?: Partial<ExtensionState>
```

Comment:
Relaxed type here, so that we could move away from using private `view` property.

See: https://github.com/RooVetGit/Roo-Code/pull/2182/files#r2022956143